### PR TITLE
Add CHANGELOG.md, also to docs

### LINKED ISSUE #1435: Changelog on rtd documentation
At least a link to the github release notes would be good. If it is somewhere already, I could not find it :)

### PR DESCRIPTION
**Pull request by yannikschaelte**
Fixes #1435. Created via

```python
#!/bin/bash

import requests
import json
import datetime

url_releases = "https://api.github.com/repos/amici-dev/amici/releases"

releases = requests.get(
    url=url_releases,
    params={'accept': 'application/vnd.github.v3+json',
            'per_page': 1000}
).json()

# Output text
out = "# Changelog \n\n"

for release in releases:
    date = datetime.datetime.strptime(
        release["published_at"], "%Y-%m-%dT%H:%M:%SZ")
    date_str = " (" + str(date.date()) + ")"

    header = "## " + release["tag_name"] + date_str
    out += header + "\n\n"

    out += release["body"].replace('## ', '### ') + "\n\n"

out = out.replace('\r', '')

with open("CHANGELOG.md", "w") as f:
    f.write(out)
```

### CODE REVIEW

**Review by FFroehlich — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,890 @@
...
+3) AMICI now also supports second order adjoint sensitivities (full (via the o2flag = 1 and as a vector product via o2flag = 2)
+4) AMICI now supports more SBML, SBML v2 and rate rules
+
+
+## 0.2.1 (2016-05-09)
+
+Bugfix release. This release also includes some changes that should improve the performance on the new R2016a release of MATLAB.
+
+
+## 0.2 (2016-03-17)
```
Comment (dweindl) [inline CHANGELOG.md:878]:
```suggestion
## v0.2 (2016-03-17)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,890 @@
...
+
+
+## v0.4.0 (2017-05-15)
+
+* First citable version of AMICI (via zenodo integration).
+* Better support for standalone compilation
+* Improved SBML import scripts
+* General Code Cleanup
+
+## 0.3.0 (2016-09-05)
```
Comment (dweindl) [inline CHANGELOG.md:863]:
```suggestion
## v0.3.0 (2016-09-05)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,890 @@
...
+- Added c++ interface
+- Added support for second order adjoint computation
+
+- Rewrote large parts of the code as proper c++11 code to allow easier code maintanance
+- Substantially extended testing in continuous integration to assure code quality
+
+## Second_order_adjoint_and_hybrid_profile_calculation_submission (2018-01-29)
+
+
+
```
Comment (dweindl) [inline CHANGELOG.md:855]:
```suggestion
```

**Review by dweindl — APPROVED** [review]
Great, thanks.

**Pull request merged**